contracts/base/LoansBase.sol

```diff
@@ -494,10 +494,57 @@ contract LoansBase is LoansInterface, Base {
         address payable recipient
     ) internal {
         if (loans[loanID].escrow != address(0x0)) {
-            EscrowInterface(loans[loanID].escrow).claimTokens(recipient);
+            /**
+                - Check the value of the liquidator will get as a reward - total amount owed * 1.05 and converted into collateral asset 
+                settings().chainlinkAggregator().valueFor(
+                    baseAddress,
+                    quoteAddress,
+                    baseAmount
+                ); */
+            uint256 totalOwed = _getTotalOwed(loanID);
+            uint256 amountToLiquidatorInLendingTokens = totalOwed.mul(105).div(100);
+            uint256 amountToLiquidatorInCollateralTokens = settings()
+                .chainlinkAggregator()
+                .valueFor(
+                lendingPool.lendingToken(),
+                collateralToken,
+                amountToLiquidatorInLendingTokens
+            );
+            uint256 leftoverTokens;
+            /** - if converted collateral value > amount 
+                    _payOutCollateral(loanID, amount, recipient);
+                    - amount of value in tokens from the escrow to be claimed = converted collateral value - amount [if escrow exists], claimTokens(receipient, converted collateral value) - else the borrower has to call the claimTokens */
+            if (amountToLiquidatorInCollateralTokens > totalOwed) {
+                _payOutCollateral(loanID, amount, recipient);
+                leftoverTokens = amountToLiquidatorInCollateralTokens - totalOwed;
+                EscrowInterface(loans[loanID].escrow).claimTokensByCollateralValue(
+                    loans[loanID].loanTerms.borrower,
+                    leftoverTokens
+                );
+            } else if (amountToLiquidatorInCollateralTokens == totalOwed) {
+                /** - if converted collateral value = amount
+                    _payOutCollateral(loanID, amount, recipient); */
+                _payOutCollateral(loanID, amount, recipient);
+            } else if (amountToLiquidatorInCollateralTokens < amount) {
+                /** - if converted collateral value < amount
+                    _payOutCollateral(loanID, convertedCollateralValue, recipient);
+                    _payOutCollateral(loanID, {amount - convertedCollateralValue}, loan.borrower);
+                    */
+                _payOutCollateral(
+                    loanID,
+                    amountToLiquidatorInCollateralTokens,
+                    recipient
+                );
+                leftoverTokens = amount - amountToLiquidatorInCollateralTokens;
+                _payOutCollateral(
+                    loanID,
+                    leftoverTokens,
+                    loans[loanID].loanTerms.borrower
+                );
+            }
```

Comment:
We should check the `amountToLiquidatorInCollateralTokens < loans[loanID].collateral` _before_ this if at the top of the function: 
```
if (loans[loanID].escrow != address(0x0)) {
```

So if we have enough collateral to payout the liquidator their reward, pay them out and skip the rest as there are no tokens to claim.

Reply:
moved